eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/CommonFundingHandlers.scala

```diff
@@ -37,11 +40,50 @@ trait CommonFundingHandlers extends CommonHandlers {
 
   this: Channel =>
 
-  def watchFundingTx(commitment: Commitment, additionalKnownSpendingTxs: Set[ByteVector32] = Set.empty): Unit = {
+  def watchFundingSpent(commitment: Commitment, additionalKnownSpendingTxs: Set[ByteVector32] = Set.empty): Unit = {
     val knownSpendingTxs = Set(commitment.localCommit.commitTxAndRemoteSig.commitTx.tx.txid, commitment.remoteCommit.txid) ++ commitment.nextRemoteCommit_opt.map(_.txid).toSet ++ additionalKnownSpendingTxs
     blockchain ! WatchFundingSpent(self, commitment.commitInput.outPoint.txid, commitment.commitInput.outPoint.index.toInt, knownSpendingTxs)
   }
 
+  def watchFundingConfirmed(fundingTxId: ByteVector32, minDepth_opt: Option[Long]): Unit = {
+    minDepth_opt match {
+      case Some(fundingMinDepth) => blockchain ! WatchFundingConfirmed(self, fundingTxId, fundingMinDepth)
+      // When using 0-conf, we make sure that the transaction was successfully published, otherwise there is a risk
+      // of accidentally double-spending it later (e.g. restarting bitcoind would remove the utxo locks).
+      case None => blockchain ! WatchPublished(self, fundingTxId)
+    }
+  }
+
+  def acceptFundingTxConfirmed(w: WatchFundingConfirmedTriggered, d: PersistentChannelData): MetaCommitments = {
+    log.info("funding txid={} was confirmed at blockHeight={} txIndex={}", w.tx.txid, w.blockHeight, w.txIndex)
+    d.metaCommitments.latest.localFundingStatus match {
+      case _: SingleFundedUnconfirmedFundingTx =>
+        // in the single-funding case, as fundee, it is the first time we see the full funding tx, we must verify that it is
+        // valid (it pays the correct amount to the correct script). We also check as funder even if it's not really useful
+        Try(Transaction.correctlySpends(d.metaCommitments.latest.fullySignedLocalCommitTx(keyManager).tx, Seq(w.tx), ScriptFlags.STANDARD_SCRIPT_VERIFY_FLAGS)) match {
+          case Success(_) => ()
+          case Failure(t) =>
+            log.error(t, s"rejecting channel with invalid funding tx: ${w.tx.bin}")
+            throw InvalidFundingTx(d.channelId)
+        }
+      case _ => () // in the dual-funding case, we have already verified the funding tx
+    }
```

Comment:
This allowed me to factor the single and dual funding cases, which I think is worth it and make the handling in the channel FSM more consistent.